packages/flutter_tools/lib/src/test/coverage_collector.dart

```diff
@@ -141,3 +154,120 @@ class CoverageCollector extends TestWatcher {
     return true;
   }
 }
+
+Future<Map<String, dynamic>> collect(Uri serviceUri, bool Function(String) libraryPredicate) async {
+  VMService vmService;
+  try {
+    vmService = await VMService.connect(serviceUri, compression: CompressionOptions.compressionOff);
+    await vmService.getVM();
+  } on TimeoutException {
+    rethrow;
+  }
+  return await _getAllCoverage(vmService, libraryPredicate);
+}
+
+
+Future<Map<String, dynamic>> _getAllCoverage(VMService service, bool Function(String) libraryPredicate) async {
+  await service.getVM();
+  final List<Map<String, dynamic>> coverage = <Map<String, dynamic>>[];
+
+  for (Isolate isolateRef in service.vm.isolates) {
+    await isolateRef.load();
+    final Map<String, dynamic> scriptList = await isolateRef.invokeRpcRaw('getScripts', params: <String, dynamic>{'isolateId': isolateRef.id});
+    final List<Future<void>> futures = <Future<void>>[];
+
+    final Map<String, Map<String, dynamic>> scripts = <String, Map<String, dynamic>>{};
+    final Map<String, Map<String, dynamic>> sourceReports = <String, Map<String, dynamic>>{};
+    for (Map<String, dynamic> script in scriptList['scripts']) {
+      if (!libraryPredicate(script['uri'])) {
+        continue;
+      }
+      final String scriptId = script['id'];
+      futures.add(
+        isolateRef.invokeRpcRaw('getSourceReport', params: <String, dynamic>{'forceCompile': true, 'scriptId': scriptId, 'isolateId': isolateRef.id, 'reports': <String>['Coverage']})
+          .then((Map<String, dynamic> report) {
+            sourceReports[scriptId] = report;
+          })
+      );
+      futures.add(
+        isolateRef.invokeRpcRaw('getObject', params: <String, dynamic>{'isolateId': isolateRef.id, 'objectId': scriptId})
+          .then((Map<String, dynamic> script) {
+            scripts[scriptId] = script;
+          })
+      );
+    }
+    await Future.wait(futures);
+    final Map<Uri, Map<int, int>> hitMaps = <Uri, Map<int, int>>{};
```

Comment:
To help with readability you might split this code up into some more functions.

Reply:
I refactored this into a method for loading and a method for building